### PR TITLE
Util to find English words on .data

### PR DESCRIPTION
**Pull request by theMiddleBlue**
this could be useful for our lists, it checks English words on .data files that could lead to FPs

### CODE REVIEW

**Review by fzipi — APPROVED** [review]
It is a nice starting script!

**Review by fzipi — CHANGES_REQUESTED** [review]
Maybe doing a shellcheck pass first?

**Comment by theMiddleBlue** [comment]
> Maybe doing a shellcheck pass first?

done

File: util/fp-finder/spell.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+RULES_DIR="$SCRIPT_DIR/../../rules/"
+
+for datafile in "$RULES_DIR"*.data; do
+    DATAFILE_NAME=${datafile##*/}
+    
+    echo "-> checking $DATAFILE_NAME"
+
+    for word in $(grep -E '^[a-z]+$' "$datafile" | uniq | sort); do
+        IS_NOT_ENGLISH=$(echo "$word" | spell | wc -l)
```
Comment (theseion) [inline util/fp-finder/spell.sh:13]:
```suggestion
        IS_NOT_ENGLISH=$(echo "${word}" | spell | wc -l)
```

File: util/fp-finder/spell.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        if [ "$IS_NOT_ENGLISH" -lt 1 ]; then
```
Comment (theseion) [inline util/fp-finder/spell.sh:14]:
```suggestion
        if [ "${IS_NOT_ENGLISH}" -lt 1 ]; then
```

File: util/fp-finder/spell.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            echo "   \`- found English word: $word"
```
Comment (theseion) [inline util/fp-finder/spell.sh:15]:
```suggestion
            echo "   \`- found English word: ${word}"
```

File: util/fp-finder/spell.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+RULES_DIR="$SCRIPT_DIR/../../rules/"
+
+for datafile in "$RULES_DIR"*.data; do
+    DATAFILE_NAME=${datafile##*/}
+    
+    echo "-> checking $DATAFILE_NAME"
```
Comment (theseion) [inline util/fp-finder/spell.sh:10]:
```suggestion
    echo "-> checking ${DATAFILE_NAME}"
```

File: util/fp-finder/spell.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+RULES_DIR="$SCRIPT_DIR/../../rules/"
```
Comment (theseion) [inline util/fp-finder/spell.sh:5]:
```suggestion
RULES_DIR="${SCRIPT_DIR}/../../rules/"
```

File: util/fp-finder/spell.sh [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+for datafile in "$RULES_DIR"*.data; do
+    DATAFILE_NAME=${datafile##*/}
+    
+    echo "-> checking $DATAFILE_NAME"
+
+    for word in $(grep -E '^[a-z]+$' "$datafile" | uniq | sort); do
```
Comment (theseion) [inline util/fp-finder/spell.sh:12]:
```suggestion
    for word in $(grep -E '^[a-z]+$' "${datafile}" | uniq | sort); do
```

**Review by theseion — APPROVED** [review]

**Comment by theseion** [comment]
Let's see what we can use it for. IMO, it should live in the toolchain at some point.

**Review by dune73 — APPROVED** [review]

**Pull request merged**